docs/Rules/MA0098.md

```diff
@@ -1 +1,135 @@
+var array = new int[10];
+var list = new List<int>();
+
+var last1 = array.Last();               // ❌ Inefficient
+var last2 = list.Last();                // ❌ Inefficient
+
+// Compliant - using indexer
+var last3 = array[array.Length - 1];    // ✅ Direct access
+var last4 = list[list.Count - 1];       // ✅ Direct access
+
+// Note: Last(predicate) is not affected by this rule
+var lastEven = array.Last(x => x % 2 == 0);    // ✅ Uses predicate, rule doesn't apply
+````
+
+## C# 8+ Index and Range Support
+
+Starting with C# 8.0 and when targeting frameworks that support the Index and Range feature (.NET Core 3.0+, .NET 5+), the analyzer can suggest using the hat operator (`^`) for accessing elements from the end:
+
+````csharp
+// C# 8+ with compatible target framework
+var array = new int[10];
+
+// The analyzer suggests using the hat operator for Last()
+var last = array[^1];                   // ✅ Equivalent to array[array.Length - 1]
+
+// For older language versions or target frameworks, it uses the traditional syntax
+var last = array[array.Length - 1];     // ✅ Fallback for older versions
+````
+
+## Performance Benefits
+
+Using indexer access instead of LINQ methods provides several performance advantages:
+
+- **Direct Access**: Indexers provide O(1) direct access to elements, while LINQ methods may involve enumeration overhead
+- **No Extension Method Overhead**: Eliminates the cost of extension method calls and delegate invocations
+- **Reduced Memory Allocations**: Avoids temporary objects and iterator allocations that LINQ methods might create
+- **Better JIT Optimization**: The JIT compiler can better optimize simple indexer access compared to generic extension methods
+
+## When This Rule Doesn't Apply
+
+This rule has specific limitations and won't trigger in certain scenarios:
+
+### Methods with Predicates
+
+The rule only applies to parameterless versions of `First()` and `Last()`, and single-parameter `ElementAt()`. It doesn't apply when using predicates:
+
+````csharp
+// These are NOT flagged by the rule - predicates are necessary
+var firstEven = list.First(x => x % 2 == 0);
+var lastNegative = array.Last(x => x < 0);
+````
+
+### OrDefault Variants
+
+The rule doesn't apply to the "OrDefault" variants of these methods:
+
+````csharp
+// These are NOT flagged by the rule
+var item1 = list.FirstOrDefault();      // ✅ Rule doesn't apply
+var item2 = list.LastOrDefault();       // ✅ Rule doesn't apply
+var item3 = list.ElementAtOrDefault(5); // ✅ Rule doesn't apply
+````
+
+### Unsupported Collection Types
+
+The rule only applies to collections that implement `IList<T>` or `IReadOnlyList<T>`:
+
+````csharp
+// These are NOT flagged by the rule - no indexer support
+IEnumerable<int> enumerable = GetNumbers();
+var first = enumerable.First();         // ✅ Rule doesn't apply
+
+HashSet<int> hashSet = new();
+var firstFromSet = hashSet.First();     // ✅ Rule doesn't apply
+````
+
+## Related Rules
```

Comment:
If we add this to this rule, it should be added to all other rules to be consistent.

Reply:
might do this in a follow up PR